code/modules/mob/living/simple_animal/guardian/guardian_creator.dm

```diff
@@ -87,9 +87,10 @@ GLOBAL_LIST_INIT(guardian_radial_images, setup_guardian_radial())
 	if(LAZYLEN(candidates))
 		var/mob/dead/observer/candidate = pick(candidates)
 		spawn_guardian(user, candidate, guardian_path)
+		used = TRUE
+		SEND_SIGNAL(src, COMSIG_ITEM_TC_USED)
```

Comment:
I am worried about this signal because it is named generically but only sent from one place, we _prefer_ signals to only be sent from one place but that means they should be named in specific to what the signal is. On the other hand, we don't want every one of these elements to register to a bunch of item-specific signals.

It would probably be better to just call `RemoveElement` here... but that won't work unless the item somehow knows what its discount was, which it does not 😓 

The third option is to turn this back into a component rather than an element, use `GetComponent`, and delete it.
Not sure what the best approach is here though to be honest.

Reply:
just turn it into a macro signal instead if you're worried about genericity and potential clashes (although I think this solution is okay) 
```dm
#define COMSIG_TRAITOR_ITEM_USED(item) "traitor_used_[item]"
```

`item` in the above example would be the typepath of the `/obj/item` we're using, and we pass the path into the element for the signal registration. for a living breathing example of this in the codebase, look at `COMSIG_AI_BLACKBOARD_KEY_SET()`

Reply:
Alright I've made the changes!